### DISCUSSION TITLE
How do I create a GitHub Action for the marketplace using Python?

### DISCUSSION THREAD

**Discussion by paramt** [community/community#25880] [Actions]
<p>I want to upload my action to the marketplace, so that whenever someone uses the action it should run a python script. This is how many action.yml looks like so far</p>
<pre><code>name: Check URLs
author: Param Thakkar
description: Automatically check for broken links in markdown files
inputs:
  markdown_files:
    description: 'A list of all the markdown files in the repo, separated by commas. Defaults to README.md'
    required: false
    default: 'README.md'
  blacklisted_links:
    description: 'A comma-separated list of URLs to skip over.'
    required: false
runs:
  using: python
  main: entrypoint.py
branding:
  icon: 'link'
  color: 'green'
</code></pre>
<p>However, I am not able to ‘Publish to Marketplace’ when creating a new release. I have GitHub actions enabled on my user account… What am I missing here?</p>
<p>P.S. my repo is private. Could that be why it’s not letting me upload to Marketplace?</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/paramt">@paramt</a>,</p>
<p>Thank you for being here! Publishing an action to the GitHub Marketplace requires that the action lives in your <strong>public repository’s root directory</strong> and it exists alone in that repository without other actions. To learn more about building GitHub Actions, see “<a href="https://developer.github.com/actions/creating-github-actions/" rel="nofollow noopener">Creating GitHub Actions</a>.”</p>

**Comment by paramt** [comment]
<p>The problem was that the repo wasn’t public. Thanks!</p>
